core/backend/src/IModelDb.ts

```diff
@@ -2524,7 +2525,7 @@ export class SnapshotDb extends IModelDb {
    * @internal
    */
   public static async openCheckpointV2(checkpoint: CheckpointProps): Promise<SnapshotDb> {
-    const filePath = await V2CheckpointManager.attach(checkpoint);
+    const { filePath, sasTokenExpiry } = await V2CheckpointManager.attach(checkpoint);

```

Comment:
it seems to me that the auto-reattach should happen *inside* `V2CheckpointManager.attach` so no caller needs to worry about it.